Gordon360/AuthorizationFilters/StateYourBusiness.cs

```diff
@@ -503,6 +553,21 @@ private bool canUpdate(string resource)
                     }
                 case Resource.STUDENT:
                     return false; // No one should be able to update a student through this API
+                case Resource.HOUSING:
+                    {
+                        // The user must be on an application and be an editor to update the application
+                        var housingService = new HousingService(new UnitOfWork());
+                        var sess_cde = Helpers.GetCurrentSession().ToString();
+                        int? applicationID = housingService.GetApplicationID(user_id, sess_cde);
+                        if (applicationID != null)
+                        {
+                            var editorID = housingService.GetEditorID(applicationID);
+                            var is_editor = editorID == user_id;
+                            if (is_editor)
+                                return true;
+                        }
+                        return false;
+                    }
```

Comment:
We need to add a check for `housing.checkHousingAdmin(user_id)` here so that the housing admin can push updates to applications, such as the confirmations we planned for probation, off-campus, or others that I don't remember right now.